underdocs-version/src/main/kotlin/underdocs/version/retriever/DefaultVersionRetriever.kt

```diff
@@ -0,0 +1,20 @@
+package underdocs.version.retriever
+
+import underdocs.version.Version
+import underdocs.version.reader.VersionPropertiesReader
+import java.util.*
+
+class DefaultVersionRetriever: VersionRetriever {
+    override fun get(): Version {
+        val versionProperties : Properties by lazy {
```

Comment:
Generally, `by lazy` should be used for class properties.

Here, using `by lazy` achieves the following: the value of the `versionProperties` variable will only be resolved once it's needed. However, it's needed right afterwards. Then, the value is just thrown away and will be determined again if `get()` is called. Thus, the laziness of `by lazy` is not really in use.

On the other hand, if you create a class property, then `by lazy` will really start to shine:

  * the value of the property will be loaded lazily: only if requested,
  * once loaded, the value will be there and subsequent loads are not necessary.

Reply:
I think I misunderstood the use of the lazy delegate. Might need some more reading in this topic.

Reply:
We can talk about this on Discord if necessary :)

Reply:
Your explanation was good and now I see how (and why) I should have done it like that in the first place. But we could talk about this, just to make sure I don't misunderstand it.

Reply:
Sure, we'll set aside time this week for this topic!